### PR TITLE
[Requirements] Bump PyYAML

### PR DESCRIPTION
**Pull request by liranbg**
Bumped pyyaml and updated code to use safe_load / safe_dump where needed
in addition, fixed function mutable arguments and removed unused test

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -54,11 +54,11 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ['3.7', '3.8', '3.9']
+        python-version: ['3.9']
```
Comment (rokatyy) [inline .github/workflows/CI.yaml:57]: So we will stop support of python 3.7 and 3.8?

**Review by rokatyy — COMMENTED** [review]
So we will stop support of python 3.7 and 3.8?

**Comment by liranbg** [comment]
> So we will stop support of python 3.7 and 3.8?

indeed. once released we will bump minor as well.

**Review by rokatyy — APPROVED** [review]
👍  LGTM, but please add to the PR description that we stop support for py3.7 and 3.8

**Pull request merged**